connectors/airwatch/src/main/java/com/vmware/connectors/airwatch/ConnConfiguration.java

```diff
@@ -65,7 +84,11 @@ public String connectorMetadata() throws IOException {
 
     @Bean
     public URI gbBaseUrl() {
-        String gbUrl = environment.getProperty("greenbox.url");
-        return fromHttpUrl(gbUrl).build().toUri();
+        try {
+            return fromHttpUrl(environment.getProperty("greenbox.url")).build().toUri();
+        } catch (IllegalArgumentException ex) {
+            logger.error("Greenbox URL config is invalid. ", ex);
+            return null;
```

Comment:
Would it be better if this method threw an exception instead of returning null? My concern is that returning null will just cause confusing errors later on. I haven't tried it though.

Reply:
Returning null makes the application fail to start, since it can't find the required bean.
I'm attaching logs to quickly see how it looks.

[gb-url-missing-logs.txt](https://github.com/vmware/connectors-workspace-one/files/2013547/gb-url-missing-logs.txt)

Reply:
Hmm.., this seems better. I'll fix it.
Then we can see custom exception and also bean creation exception in the logs.